src/components/core/pivotTable/ResizedColumnsStore.ts

```diff
@@ -138,48 +213,104 @@ export class ResizedColumnsStore {
             this.allMeasureColumnWidth = null;
         }
 
+        this.weakMeasuresColumnWidths = this.filterWeakColumnWidthItems(columnWidths);
+
+        const columnWidthItems = this.filterStrongColumnWidthItems(columnWidths);
         const columnWidthsByField = convertColumnWidthsToMap(columnWidthItems, executionResponse);
         this.manuallyResizedColumns = columnWidthsByField;
     }
 
-    private partition(
-        array: ColumnWidthItem[],
-        isValid: (elem: ColumnWidthItem) => boolean,
-    ): [ColumnWidthItem[], ColumnWidthItem[]] {
-        if (!array) {
-            return [[], []];
+    public getMatchingColumnsByMeasure(targetColumn: Column, allColumns: Column[]): Column[] {
+        const targetMeasureLocalIdentifier: string = getMappingHeaderMeasureItemLocalIdentifier(targetColumn);
+
+        if (targetMeasureLocalIdentifier) {
+            return allColumns.filter((col: Column) => {
+                const measureLocalIdentifier = getMappingHeaderMeasureItemLocalIdentifier(col);
+                return targetMeasureLocalIdentifier === measureLocalIdentifier;
+            });
         }
+        return [];
+    }
 
-        return array.reduce<[ColumnWidthItem[], ColumnWidthItem[]]>(
-            ([pass, fail], elem) => {
-                return isValid(elem) ? [[...pass, elem], fail] : [pass, [...fail, elem]];
-            },
-            [[], []],
-        );
+    public getMatchedWeakMeasuresColumnWidths(item: Column | ColDef): IWeakMeasureColumnWidthItem {
+        const measureHeaderLocalIdentifier: string = getMappingHeaderMeasureItemLocalIdentifier(item);
+
+        if (measureHeaderLocalIdentifier) {
+            return this.weakMeasuresColumnWidths[measureHeaderLocalIdentifier];
+        }
+    }
+
+    private filterAllMeasureColumnWidthItem(columnWidths: ColumnWidthItem[]): IAllMeasureColumnWidthItem {
+        if (columnWidths) {
+            return columnWidths.filter(isAllMeasureColumnWidthItem)[0];
+        }
+    }
+
+    private filterStrongColumnWidthItems(columnWidths: ColumnWidthItem[]) {
+        if (columnWidths) {
+            return columnWidths.filter(
+                item => isAttributeColumnWidthItem(item) || isMeasureColumnWidthItem(item),
+            );
+        }
+        return [];
+    }
+
+    private filterWeakColumnWidthItems(columnWidths: ColumnWidthItem[]): IWeakMeasureColumnWidthItemsMap {
+        if (columnWidths) {
+            const onlyWeakWidthItems: IWeakMeasureColumnWidthItem[] = columnWidths.reduce(
```

Comment:
Consider simplifying to `.filter(isWeakMeasureColumnWidthItem)`

Reply:
done